whois-scheduler/src/main/java/net/ripe/db/whois/scheduler/SchedulerConfig.java

```diff
@@ -0,0 +1,62 @@
+package net.ripe.db.whois.scheduler;
+
+import net.javacrumbs.shedlock.core.LockProvider;
+import net.javacrumbs.shedlock.provider.jdbctemplate.JdbcTemplateLockProvider;
+import net.javacrumbs.shedlock.spring.annotation.EnableSchedulerLock;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.EnableAspectJAutoProxy;
+import org.springframework.context.annotation.ImportResource;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+
+
+import javax.sql.DataSource;
+import java.time.Duration;
+import java.time.ZoneId;
+import java.util.TimeZone;
+import java.util.concurrent.Executor;
+
+@Configuration
+@EnableAspectJAutoProxy(proxyTargetClass = true)
+@EnableScheduling
+@EnableSchedulerLock(defaultLockAtMostFor="23h")
+@ImportResource(value = {
+        "classpath:applicationContext-api.xml",
+        "classpath:applicationContext-commons.xml"
+})
+@ComponentScan(basePackages="net.ripe.db.whois.scheduler")
+public class SchedulerConfig {
+
+    @Bean
+    public Executor taskExecutor() {
+        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
+        executor.initialize();
+        return executor;
+    }
+
+    @Bean("scheduler")
+    public TaskScheduler taskScheduler() {
+        // set properties if required
+        ThreadPoolTaskScheduler scheduler = new ThreadPoolTaskScheduler();
+        scheduler.setPoolSize(2);
+        return scheduler;
+    }
+
+    @Bean
+    public LockProvider lockProvider(@Qualifier("internalsDataSource") DataSource dataSource) {
+        return new JdbcTemplateLockProvider(
+                JdbcTemplateLockProvider.Configuration.builder()
+                        .withJdbcTemplate(new JdbcTemplate(dataSource))
+                        .withTimeZone(TimeZone.getTimeZone("Europe/Amsterdam"))
```

Comment:
I understand the jobs run in CEST but should we still be persisting UTC so dates are consistent across the db

Reply:
Also, do we need to run *all* scheduled jobs using the Amsterdam timezone?

We could assume the default (local, UTC) timezone where it doesn't matter, and where it does matter use the `zone` argument.

Reply:
Do we need to default to Amsterdam time? I think we should default to UTC and only make exceptions where necessary (using the zone argument on the `@Scheduled` annotation)

Reply:
That is indeed the intention. Default to UTC but run the midnight jobs at Amsterdam time. I have updated the configuration and confirmed the daly midnight jobs now kicks in at midnight Amsterdam time:

```

2021-04-29T00:00:00,021 INFO  [ScheduledTaskAspect] Starting scheduled task: net.ripe.db.whois.scheduler.task.acl.ResetPersonalObjectAccounting@289fff3c
2021-04-29T00:00:00,079 INFO  [ScheduledTaskAspect] Scheduled task: net.ripe.db.whois.scheduler.task.acl.ResetPersonalObjectAccounting@289fff3c took 57.34 ms
2021-04-29T00:00:00,120 INFO  [ScheduledTaskAspect] Starting scheduled task: net.ripe.db.whois.scheduler.task.acl.AutomaticPermanentBlocks@1e6a8d01
2021-04-29T00:00:00,129 INFO  [ScheduledTaskAspect] Scheduled task: net.ripe.db.whois.scheduler.task.acl.AutomaticPermanentBlocks@1e6a8d01 took 8.803 ms
```

Reply:
Also:
```
2021-04-30T00:00:00,132 INFO  [RpslObjectsExporter] Database export started
```

Reply:
OK thanks